desc/compute/_curve.py

```diff
@@ -501,6 +502,310 @@ def _x_sss_FourierXYZCurve(params, transforms, profiles, data, **kwargs):
+    name="x",
+    label="\\mathbf{x}",
+    units="m",
+    units_long="meters",
+    description="Position vector along curve",
+    dim=3,
+    params=["X", "Y", "Z", "_knots", "_period", "_method"],
+    transforms={
+        "rotmat": [],
+        "shift": [],
+    },
+    profiles=[],
+    coordinates="s",
+    data=["s"],
+    parameterization="desc.geometry.curve.SplineXYZCurve",
+    basis="basis",
+)
+def _x_SplineXYZCurve(params, transforms, profiles, data, **kwargs):
+    xq = data["s"]
+
+    Xq = interp1d(
+        xq,
+        params["_knots"],
+        params["X"],
+        method=params["_method"],
+        derivative=0,
+        period=params["_period"],
+    )
+    Yq = interp1d(
+        xq,
+        params["_knots"],
+        params["Y"],
+        method=params["_method"],
+        derivative=0,
+        period=params["_period"],
+    )
+    Zq = interp1d(
+        xq,
+        params["_knots"],
+        params["Z"],
+        method=params["_method"],
+        derivative=0,
+        period=params["_period"],
+    )
+
+    coords = jnp.stack([Xq, Yq, Zq], axis=1)
+    coords = coords @ transforms["rotmat"].T + transforms["shift"][jnp.newaxis, :]
+    if kwargs.get("basis", "rpz").lower() == "rpz":
+        coords = xyz2rpz(coords)
+    data["x"] = coords
+    return data
+
+
+@register_compute_fun(
+    name="x_s",
+    label="\\partial_{s} \\mathbf{x}",
+    units="m",
+    units_long="meters",
+    description="Position vector along curve, first derivative",
+    dim=3,
+    params=["X", "Y", "Z", "_knots", "_period", "_method"],
+    transforms={
+        "rotmat": [],
+        "shift": [],
+    },
+    profiles=[],
+    coordinates="s",
+    data=["s"],
+    parameterization="desc.geometry.curve.SplineXYZCurve",
+    basis="basis",
+)
+def _x_s_SplineXYZCurve(params, transforms, profiles, data, **kwargs):
+    xq = data["s"]
+
+    dXq = interp1d(
+        xq,
+        params["_knots"],
+        params["X"],
+        method=params["_method"],
+        derivative=1,
+        period=params["_period"],
+    )
+    dYq = interp1d(
+        xq,
+        params["_knots"],
+        params["Y"],
+        method=params["_method"],
+        derivative=1,
+        period=params["_period"],
+    )
+    dZq = interp1d(
+        xq,
+        params["_knots"],
+        params["Z"],
+        method=params["_method"],
+        derivative=1,
+        period=params["_period"],
+    )
+
+    Xq = interp1d(
+        xq,
+        params["_knots"],
+        params["X"],
+        method=params["_method"],
+        derivative=0,
+        period=params["_period"],
+    )
+    Yq = interp1d(
+        xq,
+        params["_knots"],
+        params["Y"],
+        method=params["_method"],
+        derivative=0,
+        period=params["_period"],
+    )
+    Zq = interp1d(
+        xq,
+        params["_knots"],
+        params["Z"],
+        method=params["_method"],
+        derivative=0,
+        period=params["_period"],
+    )
```

Comment:
could move this stuff into the if block so we only do the extra work when needed, or if we want to always do it its probably easier to just list x and y as dependencies?

Reply:
listing x,y as dependencies (by listing "x" as a dependency) would stilll make it possible to incorrectly use an rpz "x" if that was already present in the data dictionary when this is called. I will just move into if block